apps/user_ldap/lib/Mapping/AbstractMapping.php

```diff
@@ -242,37 +243,20 @@ public function getListOfIdsByDn(array $fdns): array {
 		$maxSlices = $totalDBParamLimit / $sliceSize;
 		$results = [];
 
-		$slice = 1;
+		$slice = 0;
 		$fdns = array_map([$this, 'getDNHash'], $fdns);
-		$fdnsSlice = count($fdns) > $sliceSize ? array_slice($fdns, 0, $sliceSize) : $fdns;
-		$qb = $this->prepareListOfIdsQuery($fdnsSlice);
+		$qb = $this->prepareListOfIdsQuery();
+		$countFns = count($fdns);
 
-		while (isset($fdnsSlice[999])) {
+		do {
 			// Oracle does not allow more than 1000 values in the IN list,
 			// but allows slicing
-			$slice++;
-			$fdnsSlice = array_slice($fdns, $sliceSize * ($slice - 1), $sliceSize);
-
-			/** @see https://github.com/vimeo/psalm/issues/4995 */
-			/** @psalm-suppress TypeDoesNotContainType */
-			if (!isset($qb)) {
-				$qb = $this->prepareListOfIdsQuery($fdnsSlice);
-				continue;
-			}
-
-			if (!empty($fdnsSlice)) {
-				$qb->orWhere($qb->expr()->in('ldap_dn_hash', $qb->createNamedParameter($fdnsSlice, QueryBuilder::PARAM_STR_ARRAY)));
-			}
+			$fdnsSlice = $countFns > $sliceSize ? array_slice($fdns, $sliceSize * $slice, $sliceSize) : $fdns;
 
-			if ($slice % $maxSlices === 0) {
-				$this->collectResultsFromListOfIdsQuery($qb, $results);
-				unset($qb);
-			}
-		}
-
-		if (isset($qb)) {
+			$qb->setParameter('hashList', $fdnsSlice, IQueryBuilder::PARAM_STR_ARRAY);
 			$this->collectResultsFromListOfIdsQuery($qb, $results);
-		}
+			$slice++;
+		} while (count($fdnsSlice) === 1000);
```

Comment:
```suggestion
		} while (count($fdnsSlice) >= 1000);
```
Or is that too paranoid?